### ISSUE TITLE
AuditStash not found in Testsuite after upgrading to 4.1.2

### ISSUE THREAD

**Issue by swiffer** [cakephp/cakephp#14889]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 4.1.2
* Platform and Target: PHPUnit 8.5.8, PHP 7.4.9

### What you did
Upgrade to 4.1.2 (from 4.1.1) - PHPUnit throws a bunch of errors because AuditStash plugin can no longer be loaded.

- https://github.com/lorenzo/audit-stash
- https://github.com/cakephp/cakephp/commit/7fc5c93f454148bae6b1e8c2b731fd3f6b01aa03#comments

### What happened
Possibly related to Cake\Core\Exception\MissingPluginException: "Plugin AuditStash could not be found."
```
#0 ..\vendor\cakephp\cakephp\src\Core\PluginCollection.php(243): Cake\Core\PluginCollection->findPath('AuditStash')
#1 ..\vendor\cakephp\cakephp\src\Http\BaseApplication.php(117): Cake\Core\PluginCollection->create('AuditStash', Array)
#2 ..\src\Application.php(78): Cake\Http\BaseApplication->addPlugin('AuditStash')
```

**Comment by othercorey** [comment]
Please post the code of how you're adding the plugin.

**Comment by swiffer** [comment]
in src/Application.php:

```php
class Application extends BaseApplication implements AuthenticationServiceProviderInterface, AuthorizationServiceProviderInterface
{
    /**
     * Load all the application configuration and bootstrap logic.
     *
     * @return void
     */
    public function bootstrap(): void
    {
        // Call parent to load bootstrap from files.
        parent::bootstrap();

        if (PHP_SAPI === 'cli') {
            $this->bootstrapCli();
        }

        /*
         * Only try to load DebugKit in development mode
         * Debug Kit should not be installed on a production system
         */
        if (Configure::read('debug')) {
            $this->addPlugin('DebugKit');
        }

        // Load more plugins here
        $this->addPlugin('AuditStash');
        ...
```

**Comment by ADmad** [comment]
Can you check after moving the loadConfig() call above the foreach loop?

https://github.com/cakephp/cakephp/blob/94ab7ac1b1e82a48bfec6c01b8ce603b01cb76a0/src/Core/PluginCollection.php#L78

**Comment by swiffer** [comment]
🤔 no, this is not fixing it for me

**Comment by othercorey** [comment]
Wouldn't that already be called in this stack:
```
#0 ..\vendor\cakephp\cakephp\src\Core\PluginCollection.php(243): Cake\Core\PluginCollection->findPath('AuditStash')
#1 ..\vendor\cakephp\cakephp\src\Http\BaseApplication.php(117): Cake\Core\PluginCollection->create('AuditStash', Array)
#2 ..\src\Application.php(78): Cake\Http\BaseApplication->addPlugin('AuditStash')
```

It sounds like the plugin config is changed between PluginCollection construction and bootstrap.

**Comment by swiffer** [comment]
I've added         

```php
        var_dump(Configure::read('plugins'));
        exit;
        throw new MissingPluginException(['plugin' => $name]);
```

to PluginCollection::findPath and it is returning NULL when the error occurs for the first time

**Comment by swiffer** [comment]
https://github.com/cakephp/cakephp/commit/703b4860b7f3b827e609bb59c16ff79860e1eb99#diff-56af3672217e52636f93d3273b3daa1f

**Comment by othercorey** [comment]
That's one of the downsides to writing code with 0 comments. :(

**Comment by swiffer** [comment]
so the testcase is not properbly testing this, it should have failed, shouldn't it?

**Comment by othercorey** [comment]
This is in the wrong order:
```
        Configure::delete('plugins');
        $plugins = new PluginCollection();
        $path = $plugins->findPath('TestPlugin');
```

Should be:
```
        $plugins = new PluginCollection();
        Configure::delete('plugins');
        $path = $plugins->findPath('TestPlugin');
```

**Comment by othercorey** [comment]
Released in 4.1.3
